### PR TITLE
[Experiment] CommunityToolkit.Extensions.DependencyInjection package

### PR DESCRIPTION
**Pull request by Sergio0694**
###  Closes https://github.com/CommunityToolkit/Labs-Windows/discussions/463

This PR adds the new `CommunityToolkit.Extensions.DependencyInjection` package, ported from the Microsoft Store.

### CODE REVIEW

**Comment by Sergio0694** [comment]
@Arlodotexe I can't build this locally, I'm getting this:

> "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(5674,5): error MSB3491: Could not write lines to file "obj\Debug\netstandard2.0\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators.csproj.FileListAbsolute.txt". Path: obj\Debug\netstandard2.0\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators.csproj.FileListAbsolute.txt exceeds the OS max path limit. The fully qualified file name must be less than 260 characters."

Any ideas? 🤔

**Comment by Arlodotexe** [comment]
@Sergio0694 Try this: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later

I'm seeing some other build errors when running locally that you may need to content with, not too familiar with these APIs:
```
Build started...
1>------ Build started: Project: CommunityToolkit.Extensions.DependencyInjection.SourceGenerators, Configuration: Debug Any CPU ------
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Extensions\ITypeSymbolExtensions.cs(28,24,28,28): error CS1061: 'ImmutableArrayBuilder<char>' does not contain a definition for 'Span' and no accessible extension method 'Span' accepting a first argument of type 'ImmutableArrayBuilder<char>' could be found (are you missing a using directive or an assembly reference?)
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Extensions\IncrementalValuesProviderExtensions.cs(62,30,62,61): error CS0306: The type 'ImmutableArrayBuilder<TElement>' may not be used as a type argument
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Extensions\IncrementalValuesProviderExtensions.cs(87,41,87,72): error CS0306: The type 'ImmutableArrayBuilder<TElement>' may not be used as a type argument
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\ServiceProviderGenerator.Execute.cs(205,111,205,115): error CS1061: 'ImmutableArrayBuilder<ArgumentSyntax>' does not contain a definition for 'Span' and no accessible extension method 'Span' accepting a first argument of type 'ImmutableArrayBuilder<ArgumentSyntax>' could be found (are you missing a using directive or an assembly reference?)
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Diagnostics\DiagnosticDescriptors.cs(68,9,68,26): warning RS2000: Rule 'SRVCNFG0004' is not part of any analyzer release
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Diagnostics\DiagnosticDescriptors.cs(23,9,23,26): warning RS2000: Rule 'SRVCNFG0001' is not part of any analyzer release
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Diagnostics\DiagnosticDescriptors.cs(53,9,53,26): warning RS2000: Rule 'SRVCNFG0003' is not part of any analyzer release
1>D:\source\uwp\lib\Labs-Windows\components\Extensions.DependencyInjection\CommunityToolkit.Extensions.DependencyInjection.SourceGenerators\Diagnostics\DiagnosticDescriptors.cs(38,9,38,26): warning RS2000: Rule 'SRVCNFG0002' is not part of any analyzer release
1>CSC : warning RS2003: Rule 'TKEXDI0001' was shipped in analyzer release '1.0', but is no longer a supported diagnostic for any analyzer. Add an entry for this rule in a 'Removed Rules' table to unshipped file.
1>CSC : warning RS2003: Rule 'TKEXDI0002' was shipped in analyzer release '1.0', but is no longer a supported diagnostic for any analyzer. Add an entry for this rule in a 'Removed Rules' table to unshipped file.
1>CSC : warning RS2003: Rule 'TKEXDI0003' was shipped in analyzer release '1.0', but is no longer a supported diagnostic for any analyzer. Add an entry for this rule in a 'Removed Rules' table to unshipped file.
1>CSC : warning RS2003: Rule 'TKEXDI0004' was shipped in analyzer release '1.0', but is no longer a supported diagnostic for any analyzer. Add an entry for this rule in a 'Removed Rules' table to unshipped file.
1>Done building project "CommunityToolkit.Extensions.DependencyInjection.SourceGenerators.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 1:41 PM and took 00.558 seconds ==========

```

**Comment by Sergio0694** [comment]
@Arlodotexe I think I got most of the issues, but the CI is still failing. This doesn't look right:

> "D:\a\Labs-Windows\Labs-Windows\tooling\GlobalUsings_Samples.cs(8,45): error CS0234: The type or namespace name 'WindowsRuntime' does not exist in the namespace 'System.Runtime.InteropServices' (are you missing an assembly reference?)"

It seems like the global usings are still trying to add some WinRT stuff? Which of course isn't available here, as the projects are only .NET Standard. Also not sure why I'm getting warnings about missing samples still, even though I suppressed them 🤔

**Comment by Arlodotexe** [comment]
Thanks @Sergio0694, I've fixed the tooling issue preventing us from building samples for this component. 

I still received one of the build errors we were seeing when UWP is enabled. WinAppSdk doesn't seem to have the same issue.
image
image

**Comment by Arlodotexe** [comment]
I wasn't able to pin down the source of the issue here. This does not appear to be the fault of your project, and the issue is not present when referenced from a normal Universal Windows library project. This is confusing, as the MultiTarget system simply converts your `<MultiTarget>` preferences into `TargetFrameworks` and the needed packages for the platform(s) you chose.

I've adjusted our tooling so we can simply remove the Sample project entirely, allowing us to ship the source project now and fix the sample issue later. 

We'll need to close the tooling PRs before we close this one.

File: components/Extensions.DependencyInjection/src/CommunityToolkit.Extensions.DependencyInjection.csproj [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<Project Sdk="MSBuild.Sdk.Extras/3.0.23">
+  <PropertyGroup>
+    <ToolkitComponentName>Extensions.DependencyInjection</ToolkitComponentName>
+    <Description>This package contains Extensions.DependencyInjection.</Description>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <Version>0.0.1</Version>
```
Comment (Arlodotexe) [inline components/Extensions.DependencyInjection/src/CommunityToolkit.Extensions.DependencyInjection.csproj:6]:
```suggestion
    <Version>8.0.0-beta.1</Version>
```

**Review by Arlodotexe — APPROVED** [review]

**Pull request merged**